### PR TITLE
Cleanup cmake threading options

### PR DESCRIPTION
**Pull request by SimonHeybrock**
- Remove negation from option (DISABLE_MULTI_THREADING becomes THREADING).
- If THREADING=ON, *require* TBB. Not requiring TBB leads to unpredictable build/packaging results, as seen recently when making pip packages.
- Remove the ENABLE_THREAD_LIMIT option. We should instead add runtime control for this.

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -102,36 +102,22 @@ add_compile_options($<$<CXX_COMPILER_ID:MSVC>:-MP>)
...
     find_package(TBB CONFIG REQUIRED)
     install(IMPORTED_RUNTIME_ARTIFACTS TBB::tbb RUNTIME DESTINATION
             ${PYTHONDIR}
     )
   else()
-    find_package(TBB CONFIG)
+    find_package(TBB CONFIG REQUIRED)
   endif()
 endif()
 if(TBB_FOUND)
```
Comment (jl-wynen) [inline lib/CMakeLists.txt:117]: Does it still make sense to have a separate check for `TBB_FOUND`? Or can this `if-else` be merged with `if(THREADING)`?

File: lib/core/CMakeLists.txt [hunk 2]
```diff
@@ -61,7 +61,7 @@ generate_export_header(${TARGET_NAME})
 target_link_libraries(
   ${TARGET_NAME} PUBLIC scipp-common scipp-units Boost::Boost Eigen3::Eigen
 )
-if(TBB_FOUND AND NOT DISABLE_MULTI_THREADING)
+if(TBB_FOUND)
```
Comment (jl-wynen) [inline lib/core/CMakeLists.txt:64]: See previous comment. Use `if(THREADING)` instead?

File: lib/CMakeLists.txt (hunk 1 above)
Comment (SimonHeybrock) [reply lib/CMakeLists.txt]: Done.

File: lib/core/CMakeLists.txt (hunk 2 above)
Comment (SimonHeybrock) [reply lib/core/CMakeLists.txt:64]: I kept this, since it is explicitly linking to TBB::tbb, so I think `if(TBB_FOUND)` is clearer and makes sense.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**